src/main/java/org/elasticsearch/index/query/RangeQueryParser.java

```diff
@@ -123,11 +123,6 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
         FieldMapper mapper = parseContext.fieldMapper(fieldName);
         if (mapper != null) {
             if (mapper instanceof DateFieldMapper) {
-                if ((from instanceof Number || to instanceof Number) && timeZone != null) {
```

Comment:
I actually found a solution to this in my latest commit and do check this in the parser now